src/Symfony/Component/HttpKernel/DependencyInjection/AddCacheWarmerPass.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+
+/**
+ * Registers the cache warmers.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class AddCacheWarmerPass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    private $tag;
+    private $serviceName;
+
+    public function __construct($serviceName = 'cache_warmer', $tag = 'kernel.cache_warmer')
+    {
+        if (2 > func_num_args()) {
```

Comment:
Yes, I see where you are getting at (it was even one of my first ideas actually), but I don't think it is really useful, and just let the deprecated error be triggered. Especially as it is silenced (with the `@` in front of the `trigger_error`), it will only be logged and that's pretty much it. So you can still continue to use with `new ...` and without args, you'll just have a little line on your logs. :}